### PR TITLE
Project4 TDD - Find an available time for a meeting

### PR DESCRIPTION
**Pull request by NaamaHolzer**
- Implement a method that receives a meeting request and returns the times available for all of the mandatory attendees of the meeting
- If there are slots available for optional attendees as well - return those slots
- Add tests to check the method's functionality

### CODE REVIEW

File: walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java [hunk 1]
```diff
@@ -30,4 +30,14 @@ public void testGreeting() {
 
     Assert.assertEquals("Hello Ada", greeting);
   }
+
+  @Test
+  public void testGreetingTrimsWhitespace() {
```
Comment (orarch) [inline walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:35]: No need for the 'test' prefix in Java tests. 

The @Test annotation denotes that. 

Best practice for tests naming is: methodUnderTest_stateBeingTested_expectedOutcome

In this case, something like "greet_nameWithWhitespaces_whitespacesTrimmed"

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,176 @@
...
+import java.util.List;
 
+// Class containing the query method to find available time ranges for a given meeting request.
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    // Find optional time ranges for the meeting.
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```
Comment (orarch) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: SMALL PRs!!

As far as I understand, you implemented the "Optional Coding Challenge", which is great, but it screams an incremental change.

I can't possibly review this code, it'd be better if you start with the simple problem and then extend it

**Review by orarch — CHANGES_REQUESTED** [review]

File: walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java (hunk 1 above)
Comment (NaamaHolzer) [reply walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:35]: These tests were actually given. We only had to write the tests that check the optional attendees functionality

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (NaamaHolzer) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: This part was actually mandatory and not the optional challenge. I felt like it was a big PR but in the tutorial it said we should  open a PR when we're done...
Anyway - I'm sorry. I'll try to see how to divide it into 2 smaller PRs.

**Pull request closed without merging**